addons/mail/static/src/models/messaging_notification_handler/messaging_notification_handler.js

```diff
@@ -731,6 +733,28 @@ function factory(dependencies) {
             }
         }
 
+        /**
+         * @private
+         * @param {Object} data
+         */
+        _handleNotificationPartnerIap(data) {
+            let notification;
+            if (data.is_credit_error) {
+                const content = `<a class='btn btn-link' href=${data.url} target='_blank' ><i class='fa fa-arrow-right'></i> ${this.env._t("Buy more credits")}</a>`;
+                notification = {
+                    title: data.title,
+                    message: content,
+                    messageIsHtml: true,
+                };
```

Comment:
As for the security part, if you use `messageIsHtml: true` you need to escape what you inject into code.

Here at least `data.url` and ideally also the translation of "Buy more credits" (though we generally don't care too much about translation injection in general).